src/main/java/seedu/duke/storage/Storage.java

```diff
@@ -15,4 +30,50 @@
 
     private Date todaysDate;
     public WhatIAteList whatIAteTodayList =  new WhatIAteList(todaysDate);
+
+    //@author swatim
+    //Loads data in the form of ArrayList<String> data from the save file
+    public ArrayList<String> loadDataFromSaveFile(String filePath) throws NullPointerException, IOException {
+        BufferedReader bufferedReader;
+        try {
+            FileReader fileReader = new FileReader(filePath);
+            bufferedReader = new BufferedReader(fileReader);
+        } catch (FileNotFoundException e) {
+            InputStream inputStream = getClass().getClassLoader().getResourceAsStream(filePath);
+            bufferedReader = new BufferedReader(new InputStreamReader(inputStream));
+        }
+        ArrayList<String> data = new ArrayList<>();
+        String line;
+        while ((line = bufferedReader.readLine()) != null) {
+            data.add(line);
+        }
+        return data;
+    }
+
+    //Write data from ArrayList<String> data onto save file
+    public void writeDataOntoSaveFile(String filePath, ArrayList<String> data) throws IOException {
+        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(filePath));
+        for (String dataLine : data) {
+            bufferedWriter.write(dataLine + '\n');
+        }
+        bufferedWriter.close();
+    }
```

Comment:
This seems to be a general query for any list, can we reuse this and load method for any .txt file?

Reply:
Yes, `writeDataOntoSaveFile()` and `loadDataFromSaveFile()` can be reused for any txt.file. Can refer to my `writeTaskList()` and `readTaskList()` methods (which are right below the above function) for an example of how it can be used.